core/unit_test/TestLocalDeepCopy.hpp

```diff
@@ -1042,4 +1042,73 @@ TEST(TEST_CATEGORY, local_deepcopy_rangepolicy_layoutright) {
   }
 }
 #endif
+
+namespace Impl {
+template <typename T, typename SHMEMTYPE>
+using ShMemView =
+    Kokkos::View<T, Kokkos::LayoutRight, SHMEMTYPE, Kokkos::MemoryUnmanaged>;
+
+struct DeepCopyScratchFunctor {
+  DeepCopyScratchFunctor(
+      Kokkos::View<double*, TEST_EXECSPACE::memory_space> check_view_1,
+      Kokkos::View<double*, TEST_EXECSPACE::memory_space> check_view_2)
+      : check_view_1_(check_view_1),
+        check_view_2_(check_view_2),
+        N_(check_view_1.extent(0)) {}
+
+  KOKKOS_INLINE_FUNCTION void operator()(
+      Kokkos::TeamPolicy<TEST_EXECSPACE,
+                         Kokkos::Schedule<Kokkos::Dynamic>>::member_type team)
+      const {
+    using ShmemType = TEST_EXECSPACE::scratch_memory_space;
+    auto shview =
+        Impl::ShMemView<double**, ShmemType>(team.team_scratch(1), N_, 1);
+
+    Kokkos::parallel_for(
+        Kokkos::TeamThreadRange(team, N_), KOKKOS_LAMBDA(const size_t& index) {
+          auto thread_shview = Kokkos::subview(shview, index, Kokkos::ALL());
+          Kokkos::Experimental::local_deep_copy(thread_shview, index);
+        });
+    Kokkos::Experimental::local_deep_copy(
+        check_view_1_, Kokkos::subview(shview, Kokkos::ALL(), 0));
+
+    Kokkos::Experimental::local_deep_copy(team, shview, 6.);
+    Kokkos::Experimental::local_deep_copy(
+        check_view_2_, Kokkos::subview(shview, Kokkos::ALL(), 0));
```

Comment:
every thread executes this thing individually right now. pass the team handle here or restrict with a single clause. In fact currently you got in principle a race between the two local deep copies here. I.e. thread 1 may copy a only partly filled with 6es shview into check_view_2_. In fact since there is no barrier, its not clear that thread 2 will see any of the writes of thread 1 that is also true for the previous deep copy. Though we may wanna have the team_handle version fence I guess ...